src/util/requester.ts

```diff
@@ -81,16 +78,28 @@ interface RequesterResult<T> {
 }
 
 /**
- * Centralized management of outbound http requests.
- * Enforces rate limiting on a single instance (complying with the N Readers - 1 Writer arch for EA scaling)
- * by adding requests into a queue, processing them sequentially and sleeping when it reaches its limit.
- * The queue will throw an error if the Requester attempts to add more items than the max configured.
- * It additionally serves to coalesce requests by utilizing a more complex queue structure:
- *   - ignores duplicate items via a provided key
- *   - doesn't use the request itself because it's common for those to have things like timestamps/nonces
+ * Manages outbound HTTP requests with queueing, rate limiting, and connection reuse.
```

Comment:
This one I would honestly remove entirely. It adds complexity but it's only useful for outbound http requests, and we're not really doing that many. Most LL adapters use a streaming protocol anyways, so this should not affect performance at all.